### ISSUE TITLE
Eliminate minute and second from footprint

### ISSUE THREAD

**Issue by BaNazari** [marleymarl/geotime#99]
Most of people can not remember the exact time of their presence in a place. We force them to input minute and second and we use this exact amounts in "check my footprint" section. I think it is better (and safer) to only take time entry as hour and then add a safety factor or constant minus duration, so it covers for example from half an hour before real entry.

**Comment by marleymarl** [comment]
I agree on the seconds but disagree on the minutes. Per hour is not precise enough.

**Comment by heldersepu** [comment]
I agree with the seconds that is not really needed but minutes even if not precise do help...
Will send a fix for the seconds shortly

**Comment by BaNazari** [comment]
How about a "half-hour" unit? Or quarter? I think it's more compatible with how human remembers his actions. Also this we the process of time matching would be more "correct", as both patient and "check my footprint user" enter data within time boxes which are closer to what really has happened. I mean in a situation that the patient has entered 2:35 and the ordinary user wants to check 2:30, it is not safe to tell the user that you were not in contact. People can not be taken as a source of truth for minutes of their presence ... 
And also, presence is not a "moment", it is a duration. We take it as a point in time.
Time box entry helps the both issues.

**Comment by heldersepu** [comment]
The situation you present:
>  patient has entered 2:35 and the ordinary user wants to check 2:30

We will not asses risk by matching the time to a precise minute, there will be time boundaries, like a margin or error in math, safer to report false positives than miss a potential contact.

**Comment by heldersepu** [comment]
Since we are dealing with a virus a margin or error could be half life on solid surfaces, so the example you presented 
> patient has entered 2:35

We could flag any one near the same location as:
- a high risk from 1:35 to 6:35
- a medium risk from 6:35 to 8:35
- a low risk from 8:35 to 10:35

**Comment by heldersepu** [comment]
Now from a technical perspective the `TimePicker` does have an interval option `minuteStep`

Here is how that looks like in action:  
![Screenshot from 2020-05-03 16-38-10](https://user-images.githubusercontent.com/545474/80925311-b1ef9e80-8d5c-11ea-8e64-4736e9846f20.png)

The selection has less items on the list, but end user is not limited instead of selecting we can still type a precise minute if desired.

You can try it here:
https://codesandbox.io/s/interval-option-ant-design-demo-hsrpx?file=/index.js

**Comment by BaNazari** [comment]
"We could flag anyone near the same location as:

a high risk from 1:35 to 6:35
a medium risk from 6:35 to 8:35
a low risk from 8:35 to 10:35"

=> Makes sense, more precise steps will be extracted from real data if needed.



"The selection has less items on the list, but end user is not limited instead of selecting we can still type a precise minute if desired."

=>With those constants I think we can leave the numbers to the user memory and focus on our estimation model. But I think time entry as a point does not work at least for most situations. People do not appear and disappear, they stay somewhere for a while. I suggest ant Range Picker.

**Comment by marleymarl** [comment]
I agree with @heldersepu - let's make it precise as possible given UX constraints (it was annonying to grab seconds) and then leave it up to the data processing layer to decide how to build rules for time ranges when using the data that has been stored.

**Comment by BaNazari** [comment]
@marleymarl  I agreed with him as well in last comment. I suggested using time ranges for another reason (people "spend time" in places).

**Comment by marleymarl** [comment]
That's an interesting concept adding ranges for each point. In essence it creates two points, one at the beginning and one at the end. There is a trade-off between additional accuracy and additional user load. I guess we can try it as an option and see how it works in practice.
